### ISSUE TITLE
Since maintanance, damage to quest boss doesn't work

### ISSUE THREAD

**Issue by Ulugbeg** [HabitRPG/habitica#7470]
Hi!

Ever since the update, the damage that I'm supposed to do to the boss doesn't work anymore, and I get error messages when I start habitica for the first time that day. This is the case for BOTH the android app (fully updated to the latest version in the App Store) and the website.

My user UUID: 3ae12e97-6fca-484e-9c92-a308e39afbb8

I tried this on different days on different platforms. On Sunday & Monday, I tried using the app (after updating it), and it said something along the lines of 'Daily verification failed' - but I didn't manage to capture a screenshot of it.

This morning, on the website, I got, I think, 3 error messages regarding invalid UUID's, but again, they faded quite quickly.

In all cases, I haven't dealt any damage to my party's boss, while I should have. In the evenings, I verified the amounts of damage I would do with the Habitica Data Display tool, and there was definately damage to be dealt :)

If you want, I can give console output tomorrow for the website - I wasn't properly prepped for that this morning :)

And, last but not least: thank you for all your hard work!

**Comment by Alys** [comment]
Similar problem: https://github.com/HabitRPG/habitrpg/issues/7411#issuecomment-220958786

For both, I can't see anything wrong in the way the quest is set up.

@Ulugbeg Your information about 'Daily verification failed' is interesting! There might be some misconfiguration in your Dailies that is preventing cron from completing correctly. I'll look through them to see if I can find it (I can do that without looking at the text of the Dailies).

I very much would like console output tomorrow if you don't mind! If we haven't solved it by then. `cross fingers`

Have you noticed anything else odd about the cron process (the reset process at the start of the day)? For example, have any of your completed Dailies NOT been changed to incomplete in the morning? If so, can you tell me which ones, if the text isn't private?

**Comment by Ulugbeg** [comment]
Thank you for your lightingfast responds ;)

Feel free to see the texts of anything in my tasks, nothing too private in there, and it's in Dutch anyway :)

Some more (potentially relevant) information: 
-  there was a challenge I was in. The creator of the challenge tried to change the day of one of the existing dailies in the challenge, that seemed to have failed/not work. This was Sunday evening, so after the maintanance/upgrade. If the ID in the A HREF tag is anything to go by, the challenge in question is 5744759b-5819-46f5-82cb-8b3a1161d5fe - it's called "Oren Hond". 
- We decided to BOTH leave the challenge, and I tried on my end to remove the daily as opposed to leaving it. I seem to remember it gave AN error, but it was late at nights and I don't quite remember what it was :) The dailies DO seem to be gone from my daily list atm.
- If I currently go to challenges, and click 'Membership -> Participating', the challenge is still there, even though it does say '0 participants' and I'm not the creator of it.

It could be that after the update, we didn't log in again or F5 the page, and that the tab for habitica was still open, so the website itself might have still been running some old, cached code.

As for your question regarding cron - the only odd thing I noticed was that at Sunday evening, local time, after midnight but before my custom daystart, it did seem that the app (again, latest version) did cron prematurely and show me the dailies for Monday. It might have been the case that I checked habitrpg (eh, habitica, old habit :)) for the first time AFTER midnight local time, but definately before my custom day start. I might be wrong on this, as I seem to remember that when I checked the website again it did show the dailies from Sunday. Come to think of it, I'm not sure if I refreshed. So, all in all: some oddness happened, don't remember the specifics.

I did not notice that weirdness from Monday -> Tuesday, but as a precautionary measure, I used the website rather than the app on then.

Oh, I didn't take any damage from the boss either, which I definitely should have, as I didn't have all dailies completed.

If the problem persists, I'll definitely add the console output too.

**Comment by Alys** [comment]
GREAT details, thank you!

Dutch will be a very effective form of encryption. ;)

The app won't be causing you to cron early - I'm certain that's impossible because cron is fully controlled by the server. The only way the app could affect when cron runs would be if it reported the wrong time zone for you, but we believe the code for reading your time zone is correct. However I have seen a couple of other reports that the app is incorrectly showing Dailies as being due when they aren't because it's not paying attention to your Custom Day Start time - that's purely a display issue (although definitely something we'll fix because it's misleading you about what Dailies are due).  So, the app is safe to use, as long as you're sure you know which Dailies are actually due. 

Not taking damage from the boss has been intentional. If you look at the quest progress messages in party chat, you'll see that the boss was asleep. ;)  Over the migration weekend, we turned off the damaging effects of cron to prevent players being unfairly affected when they couldn't log in. If you want details, see "Cron Safe Mode" near the bottom of [Cron](http://habitica.wikia.com/wiki/Cron#Cron_Safe_Mode) on the wiki. We've actually just turned off Safe Mode, but we are still preventing the boss from doing damage because there's a bug that can cause the boss to attack twice at once, so you'll keep on not getting damage from the boss for maybe a couple more days.

I'm going to look through your tasks to see if anything's wrong - that information about the challenge is probably very relevant. Thank you!

**Comment by Alys** [comment]
The "Oren Hond" challenge was still listed in your account's list of joined challenges, even though all the tasks had been removed. I've removed it now. You should no longer show up as a participant. 

Its task ":poodle: :ear::point_left: Oren hond doen" does not appear in your account - i.e., there's no task of exactly that name, and there are no tasks that contain invisible back-end data linking them to the challenge.  (Task ID 6f18ebff-1e0e-4db8-a713-608cb31b1901)

However, there is one task in your Dailies called "Oren hond doen" with no emoji. Do you recall if it was originally the task from the challenge but you edited it to remove the emoji? Does that task actually appear in your list of Dailies? (Task ID 0b92cf38-2ba3-45d5-913e-3bf988fdb73c)

It has something unusual about it in its record in the database: it does not have a 'challenge' object, whereas every task should have a 'challenge' object, even if that object is empty (as the object should be when the task is not part of the challenge). I don't know if that makes any difference, but I'm wondering if it's causing the problems you reported.

I can easily add the empty 'challenge' object to that task, and I will, but first I'd really like to know if it is currently visible in your Dailies (either due or not due - doesn't matter which, but please check both).

**Comment by Ulugbeg** [comment]
Thank you so much for looking into this, and I'm happy I'm reporting the (apparently) relevant details :)

Re: cron: yeah, I knew it was serverside, I just thought it might have happened due to the app maybe passing wrong data, but good to know that it's only a visual thing

Also, my other party member just did damage to the boss twice, killing the boss - is that the same thing you mentioned regarding /it/ doing damage twice?

I can confirm I'm no longer in the challenge, thanks!

And yes, the one with the icons is not currently in my list, so that's all good now. 
The other, non-emoji one, /is/ in my list. I think I added it Sunday night, through the website, when we were trying to get rid of the old one, from the challenge. I did still want the task :)

Also, I really appreciate you give me the backgroundinfo on what went/is wrong :)

**Comment by Alys** [comment]
Yes, that's the double quest damage bug. No need for your party mate to report it now.

Sounds like the missing 'challenge' object isn't causing any problems. It was a long stretch anyway. :)

For the moment, I'm out of ideas (I speak only for myself here, not for the other devs), but if you can get the console logs, that might give some clues!

**Comment by Ulugbeg** [comment]
Will get you those in about 17 hours :) Thank you!

**Comment by Ulugbeg** [comment]
[habitica_console_output.txt](https://github.com/HabitRPG/habitrpg/files/281233/habitica_console_output.txt)

Hello!

Well, I'm not sure this will help :( I TRIED to pay attention, but I don't think I saw ANY popups this morning when I F5'ed, but I also did NOT do any damage again to the boss.

I'm using (the latest) firefox, and in the debug console I had both the 'net' and 'js' output types selected, with (for JS), errors, warning and log ticked, but I don't see anything potentially relevant in there :(

So I'm a bit at a loss as to what I can do to help, other than try again in 24 hrs, unless there is hidden treasure in this log file of mine ;)

And yes, #7487 seems like the error message I had (well, the text, not the json bit, I didn't see that pop up anywhere).

Sorry I can't give you more that this at the moment!

**Comment by Alys** [comment]
No worries, we'll keep thinking about it.

There's something I should have checked in the beginning though: Are you
resting in the inn each night? Or clicking the "pause my dailies" button in
the Tavern in the mobile app?

**Comment by Ulugbeg** [comment]
Nope, I'm not zzzing ;)

**Comment by paglias** [comment]
> It has something unusual about it in its record in the database: it does not have a 'challenge' object, whereas every task should have a 'challenge' object, even if that object is empty (as the object should be when the task is not part of the challenge). I don't know if that makes any difference, but I'm wondering if it's causing the problems you reported.

This is causing problems for sure. But we have to understand where it happened

**Comment by paglias** [comment]
@Ulugbeg if you're able to replicate the daily validation failed messages can you open the console before it happens and from the network tab report any request marked in red? Especially the response

**Comment by Ulugbeg** [comment]
I will do that. Now, I just realised that WHY I might not have gotten the errors  today - the task is not due on wednesdays, it is on tuesdays. Shall I set it's due days also to thursdays, in the hope that I do get it again?

**Comment by paglias** [comment]
@Ulugbeg good catch! the task is ""🐩 👂👈 Oren hond doen" "?

**Comment by Ulugbeg** [comment]
Well, no, it's the Oren Hond doen WITHOUT the emoji's. Shall I indeed set it to run on Thursdays AND Fridays?  The one WITH the emoji's should be no longer in my lists, as I'm no longer in that challange. The one without emoji's is the new one which if I understood Alys correctly, is the 'weird' one.

I would guess that NOW ticking it for Wednesday's would kind of invalidate it (as it wasn't set when it was doing cron), and this way, we can see what happens:
- if it WASN'T set on one day, BUT it is on the next
- it WAS set on one day AND set on the next
- if WAS set on one day and is NOT set on the next

How does that sound?

**Comment by Ulugbeg** [comment]
Just to confirm, I set the task to happen on Thursdays & Fridays, like I proposed in my previous comment. Will report back tomorrow!

**Comment by Ulugbeg** [comment]
Hi!
I had a browser mishap which caused my sync/cron to happen before I had the console open (left the tab open yesterday, so it was the first tab it opened this morning....). I had 4 error messages, saying 'Invalid UUID', as I did manage to bash printscreen :)
However, there was nothing in the log once I did manage to open that.
So it seems as if having the 'Oren Hond'-no-emoji edition-task due on a day, causes issues :)

I'll try to make sure I DO get the console output tomorrow :( Sorry that I let you down guys & gals!

**Comment by Alys** [comment]
Don't feel bad! You're doing fine!

**Comment by ghost** [comment]
Hi!
I have similar problem. I am not able to damage the boss since the Sunday. I tried on two different computers (one with Windows and second with openSUSE). But my friend from the party is able to damage the boss. We even tried to start the mission one more time, but the problem is still present. I had error messages: "Invalid uuid", but now, after clearing cache, I don't have any message, but the problem is still present. I don't see any other problems, everything seems to be fine. Group's ID: 47a5da2e-4b7f-49cf-abc5-a260b69981fc, and mine: 0435efca-c728-4e33-84a7-bbfd60c2ad80. I will be grateful for help.

**Comment by Ulugbeg** [comment]
Ow, forgot to mention: still no damage to boss done by me. Well, I should have done damage, but the damage didn't get applied

**Comment by paglias** [comment]
@Ulugbeg I'm so sorry about this. I made a change to your data that hopefully will make it working, can you tell me if at the next cron it works?

**Comment by Ulugbeg** [comment]
Hi!
Nope didn't work, I still haven't done any damage to the boss - and it seems as if that 'Oren Hond' thing was a big ol' red herring :fish:, unless the following problem is unrelated. Sorry if this is actually a different bug, if so, I'll make another issue:

Yesterday evening, I tried to complete the task 'Gitaar Oefenen' (ID = task-d0412b20-9ab6-488d-806b-867aa8ee0cd4 according to the HTML sourcecode). It disappeared briefly, and I got the/an 'invalid UUID' error (popped up twice) and the task came back - unchecked.

This morning, the first time I opened Habitica, I got the same error message, also twice. In the console output, it seems as if the only suspicious thing in there is the following:

GET [XHR] https://habitica.com/api/v3/user/  --  HTTP/1.1 400:  Bad Request

Unf. I didn't catch this in the Firefox developer tool, just in the small 'Browser Console', so I can't see more data there.

BUT: if I (with the developer inspector thing open) try to complete the 'Gitaar Oefenen' task, I am able to get more information about that;

It seems to do an api call like so:
https://habitica.com/api/v3/tasks/d0412b20-9ab6-488d-806b-867aa8ee0cd4/score/up?userV=174

that call returns a 400/Bad Request too; and for this, I CAN inspect the json responds. And look what I find in there:

{"success":false,"error":"BadRequest","message":"**daily validation failed**","errors":[{"message":"Invalid uuid.","path":"id","value":""},{"message":"Invalid uuid.","path":"id","value":""}]}

Here's the request header used to get this responds, with my api key removed. I also removed amplitude_idhabitica.com - as I wasn't sure if posting that here would be a good OR bad idea:

Host: habitica.com
User-Agent: Mozilla/5.0 (Windows NT 6.1; WOW64; rv:46.0) Gecko/20100101 Firefox/46.0
Accept: application/json, text/plain, _/_
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate, br
DNT: 1
x-api-user: 3ae12e97-6fca-484e-9c92-a308e39afbb8
x-api-key: [redacted]
x-client: habitica-web
x-user-timezoneOffset: -120
Referer: https://habitica.com/
Cookie: session-set=true; 
Connection: keep-alive
Content-Length: 0

And this is nicely reproducable, no need to wait for crons. So, is this helpful, or is this a completely seperate issue?
